src/cogent3/core/new_alignment.py

```diff
@@ -4551,6 +4562,17 @@ def count_gaps_per_seq(
 
         return result
 
+    def count_ambiguous_per_seq(self) -> DictArray:
+        """Return the counts of ambiguous characters per sequence as a DictArray."""
+
+        darr = DictArrayTemplate(self.names)
```

Comment:
just use the `DictArray.from_array_names()` class method rather than the template class.